### PR TITLE
Document sealed_value_optional

### PR DESCRIPTION
**Pull request by sideeffffect**
document
https://github.com/scalapb/ScalaPB/issues/633
https://github.com/scalapb/ScalaPB/commit/3e24e93d9a4ae89b8d552fcad67474bbbaebe189

### CODE REVIEW

File: docs/src/main/markdown/sealed-oneofs.md [hunk 1]
```diff
@@ -91,3 +91,36 @@ Experimental Status
...
 
 Currently, sealed oneofs are implemented as a custom type defined over the old-style container message. This implementation detail is exposed through `asMessage` which returns the underlying message representing the sealed oneof.  It is possible that in a future version, sealed oneofs would have a direct implementation, and therefore `asMessage` and its return type should be considered an experimental API.
+
+Sealed oneof optional
+=====================
+
+This is a variant of _Sealed oneof_, where the optionality is expressed differently.
+The `Empty` case is not generated, but instead the sealed trait is put in other classes as `Option[_]`, not directly.
```
Comment (thesamet) [inline docs/src/main/markdown/sealed-oneofs.md:99]: To create an optional sealed oneof, name the oneof `sealed_value_optional` as in the example below:

File: docs/src/main/markdown/sealed-oneofs.md [hunk 2]
```diff
@@ -91,3 +91,36 @@ Experimental Status
 Some of the rules above are inherently required (for example, that the message types need to be distinct). Other rules, such as the one requesting that all involved messages need to be top-level, were added to make the implementation simpler. That particular rule helps ensuring that all the cases can be generated into a single Scala source file without changing too much the existing way the code generator works. It is possible that some of the rules will change over time, though most likely they are only going to become less restrictive so existing code does not break.
 
 Currently, sealed oneofs are implemented as a custom type defined over the old-style container message. This implementation detail is exposed through `asMessage` which returns the underlying message representing the sealed oneof.  It is possible that in a future version, sealed oneofs would have a direct implementation, and therefore `asMessage` and its return type should be considered an experimental API.
+
+Sealed oneof optional
```
Comment (thesamet) [inline docs/src/main/markdown/sealed-oneofs.md:95]: Optional sealed oneof

File: docs/src/main/markdown/sealed-oneofs.md (hunk 1 above)
Comment (sideeffffect) [reply docs/src/main/markdown/sealed-oneofs.md:99]: :heavy_check_mark:

File: docs/src/main/markdown/sealed-oneofs.md (hunk 2 above)
Comment (sideeffffect) [reply docs/src/main/markdown/sealed-oneofs.md]: :heavy_check_mark:

**Comment by sideeffffect** [comment]
I've amended the documentation as suggested

**Pull request merged**